mqt/qecc/cc_decoder/decoder.py

```
@@ -155,40 +156,58 @@ def simulate_error_rate(distance: int, error_rate: float, nr_sims: int, solver_p
 
+    logical_error_rates = np.full(len(code.L), 0.0)
+    logical_error_rate_ebs = np.full(len(code.L), 0.0)
+    for ler in range(len(logical_error_rates)):
+        logical_error_rates[ler] = logical_errors[ler] / nr_sims
+        logical_error_rate_ebs[ler] = np.sqrt((1 - logical_error_rates[ler]) * logical_error_rates[ler] / nr_sims)
     avg_total_time = avg_constr_time + avg_solve_time
-    logical_error_rate = logical_errors / nr_sims
-    logical_error_rate_eb = np.sqrt((1 - logical_error_rate) * logical_error_rate / nr_sims)
 
     return {
-        "distance": distance,
+        "distance": code.distance,
         "p": error_rate,
-        "logical_error_rate": logical_error_rate,
-        "logical_error_rate_eb": logical_error_rate_eb,
+        "logical_error_rates": logical_error_rates.tolist(),
+        "logical_error_rate_ebs": logical_error_rate_ebs.tolist(),
         "preconstr_time": preconstr_time.microseconds,
         "avg_constr_time": avg_constr_time,
         "avg_solve_time": avg_solve_time,
         "avg_total_time": avg_total_time,
-        "min_wt_logical_err": min_wt_logical,
+        "min_wts_logical_err": min_wt_logicals.tolist(),
     }
 
 
 def run(
-    distance: int, error_rate: float, nr_sims: int = 10000, results_dir: str = "./results_maxsat", solver: str = "z3"
+    lattice: str,
+    distance: int,
+    error_rate: float,
+    nr_sims: int = 10000,
+    results_dir: str = "./results_maxsat",
+    solver: str = "z3",
 ) -> None:
     """Run the decoding simulation for a given distance and error rate."""
-    data = simulate_error_rate(distance, error_rate, nr_sims, solver)
+    code: ColorCode
+    if lattice == str(LatticeType.HEXAGON.value):
+        code = HexagonalColorCode(distance)
+    elif lattice == str(LatticeType.SQUARE_OCTAGON.value):
+        code = SquareOctagonColorCode(distance)
+    else:
+        print("Unknown code lattice type: " + str(lattice))
+        return
```

You could extract that into a function that takes a string and returns a `ColorCode`.
That function could be placed in the `__init__.py` file for example.

With the changes to the enum, you should also not need these `str`, `value` constructs.

instead of just returning in case of an invalid code, this should throw an exception. probably `InvalidArgument`